xla/hlo/ir/hlo_instruction.cc

```diff
@@ -2074,6 +2074,38 @@ bool HloInstruction::HasSideEffect() const {
          execution_threads_set.contains(execution_thread);
 }
 
+void HloInstruction::AddSuffixToInstructionName(const std::string& suffix) {
+  // If an instruction is cloned multiple times avoid names like
+  // foo.suffix.suffix.suffix. Instead of repeating the suffix add a numeric
+  // suffix. Specifically, the clone of foo.suffix is named foo.suffix2, the
+  // clone of foo.suffix2 is named foo.suffix3 and so on.
+  const std::string dot_suffix = "." + suffix;
+  size_t index = name().rfind(dot_suffix);
+  if (index == std::string::npos) {
```

Comment:
The logic seems relatively complicated. Maybe a simple regexp match would do instead? It still seems like it should be expressible in a couple of lines, something like (in pseudocode)

```
p = r".*\.suffix(\d)$"
if (Match m = Matches(name, p)) {
  return StrCat(name.substring(m.start()), ".suffix", StrToInt(m.group(1)) + 1)
}
```

Reply:
Though seems like this code was copied, then it's fine.